trunk/source/TraceAbstraction/src/de/uni_freiburg/informatik/ultimate/plugins/generator/traceabstraction/tracehandling/StrategyFactory.java

```diff
@@ -129,31 +132,51 @@ public PathProgramCache<L> getPathProgramCache() {
 
 	/**
 	 * Constructs a {@link IRefinementStrategy} that can be used in conjunction with a {@link IRefinementEngine}.
+	 *
+	 * @param counterexample
+	 *            A trace that will be checked for feasibility and for which, if it is infeasible, a refinement result
+	 *            will be constructed.
+	 * @param controlConfigurationSequence
+	 *            Optionally, the sequence of control configurations visited by the trace in the program that is being
+	 *            verified. Used to judge the quality of proofs ("perfect") and for assert order modulation.
+	 * @param abstraction
+	 *            TODO ???
```

Comment:
Various strategies require the initial abstraction, e.g., to extract the complete alphabet, or to perform more complex generalizations.

Reply:
Ah, interesting, this is supposed to be the initial abstraction? I thought it should be the current abstraction. Good to know, I'll document it :+1: